scheduler/src/cook/mesos/api.clj

```diff
@@ -1552,21 +1558,30 @@
                                                                           (map first)
                                                                           (map str)
                                                                           (into [])))}]
-
                          :else
-                         (let [groups (mapv #(validate-and-munge-group (db conn) %) groups)
-                               jobs (mapv #(validate-and-munge-job
-                                             (db conn)
-                                             user
-                                             task-constraints
-                                             gpu-enabled?
-                                             (set (map :uuid groups))
-                                             %
-                                             :override-group-immutability?
-                                             override-group-immutability?) jobs)]
-                           [false {::groups groups
-                                   ::jobs jobs
-                                   ::pool pool}]))
+                         (let [time-until (ratelimit/time-until-millis! job-submission-rate-limiter user)
```

Comment:
The `!` at the end of function names usually implies side effects (at least to me), but the name implies it's just returning a value

Reply:
It does have side effects, and its returning a value. No different than swap!  (See https://clojuredocs.org/clojure.core/swap! )

Reply:
Makes sense. The difference is that the word "swap" implies to the caller that he's making a change, whereas "time until millis" doesn't

Reply:
Is this resolved to your satisfaction?